core/src/services/alluxio/writer.rs

```diff
@@ -0,0 +1,97 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::sync::Arc;
+
+use async_trait::async_trait;
+
+use crate::raw::*;
+use crate::*;
+
+use super::core::AlluxioCore;
+
+pub type AlluxioWriters = oio::MultipartUploadWriter<AlluxioWriter>;
+
+pub struct AlluxioWriter {
+    core: Arc<AlluxioCore>,
+
+    _op: OpWrite,
+    path: String,
+}
+
+impl AlluxioWriter {
+    pub fn new(core: Arc<AlluxioCore>, _op: OpWrite, path: String) -> Self {
+        AlluxioWriter { core, _op, path }
+    }
+}
+
+#[async_trait]
+impl oio::MultipartUploadWrite for AlluxioWriter {
```

Comment:
This is not `MultipartUploadWrite`. Please take a look over services fs.

Reply:
I'm curious? Why not MultipartUploadWrite, alluxio service is also a network request and not a local file

Reply:
`MultipartUploadWrite` means APIs like AWS S3 Multipart Uploads that support `Abort` safely and `Write` concurrently (unimplemented yet). Allxio doesn't support this semantic. It does a network request but it's behvaior is more like a POSIX fs, we should implement it by holding the `stream_id` and keep calling `write` on it.

Reply:
> It does a network request but it's behvaior is more like a POSIX fs, we should implement it by holding the `stream_id` and keep calling `write` on it.

This applies to `read` operations, but we can implement this in later PRs.

Reply:
I see. Thank you for your explanation

Reply:
AlluxioCore write is an asynchronous function, how do I call it in poll_write? I'm sorry I'm not good at this

Reply:
> AlluxioCore write is an asynchronous function, how do I call it in poll_write? I'm sorry I'm not good at this

Don't worry, we don't need to finsih all the task in one PR. Let's implement `OneshotWrite` first and polish this part later.

Reply:
plz review again